lib/modules/manager/pip-compile/extract.ts

```diff
@@ -75,17 +77,28 @@ export async function extractAllPackageFiles(
       logger.debug(`pip-compile: no content found for fileMatch ${fileMatch}`);
       continue;
     }
+    // TODO(not7cd): rename to headerArguments
     let pipCompileArgs: PipCompileArgs;
     try {
       pipCompileArgs = extractHeaderCommand(fileContent, fileMatch);
-      lockFileArgs.set(fileMatch, pipCompileArgs);
     } catch (error) {
       logger.warn(
         { fileMatch, error },
         'pip-compile: Failed to extract and parse command in output file header',
       );
       continue;
     }
+    const compileDir: string = pipCompileArgs.commandExecDir;
+    try {
+      ensureLocalPath(compileDir);
+    } catch (error) {
+      logger.warn(
+        { fileMatch, compileDir, error },
+        'pip-compile: Output file path outside of repository',
+      );
+      continue;
+    }
```

Comment:
Is this logic purely to see if the location is "local"?

Reply:
Yes